crawler/features_crawler.py

```diff
@@ -1298,6 +1301,42 @@ def _crawl_cpuHw(self):
         feature_key = '{0}-{1}'.format('cpu', cpu.cpu_vendor_id)
         yield (feature_key, cpu)
 
+
+    def crawl_gpu(self):
+        '''
+        nvidia-smi returns following: MEMORY, UTILIZATION, ECC, TEMPERATURE, 
+        POWER, CLOCK, COMPUTE, PIDS, PERFORMANCE, SUPPORTED_CLOCKS, PAGE_RETIREMENT, 
+        ACCOUNTING
+
+        currently, following are requested based on dlaas requirements: utilization.gpu
+        utilization.memory, memory.total, memory.free, memory.used
+        nvidia-smi --query-gpu=utilization.gpu,utilization.memory,memory.total,memory.free,memory.used --format=csv,noheader,nounits
+        '''
+
+        util_atttibutes = ['gpu','memory']
+        memory_atttibutes = ['total','free','used']
+
+        if not os.path.exists(NVIDIA_SMI):
+            return 
+
+        nvidia_smi_proc = subprocess.Popen([NVIDIA_SMI, '--query-gpu=utilization.gpu,utilization.memory,memory.total,memory.free,memory.used',
```

Comment:
@ricarkol: do we have a preference for subprocess vs. nvidia management library pyNVML / nvidia-ml-py ?